src/Symfony/Component/Console/Tests/phpt/single_application/version_default_name.phpt

```diff
@@ -0,0 +1,26 @@
+--TEST--
+Single Application can be executed
+--ARGS--
+--version
+--FILE--
+<?php
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\SingleCommandApplication;
+
+$vendor = __DIR__;
+while (!file_exists($vendor.'/vendor')) {
+    $vendor = dirname($vendor);
+}
+require $vendor.'/vendor/autoload.php';
+
+(new SingleCommandApplication())
+    ->setName('My Super Command')
+    ->setVersion('1.0.0')
```

Comment:
what about move the version data to the constructor? because you already can set the name there, right? and the `setVersion` method will no longer be necessary.

Reply:
I want to let the constructor open to dependency injection.

Reply:
Looks fine to be like this. Most of the time, the version is not useful (at least for proprietary tools).